### ISSUE TITLE
unarchive: no support for .bz2 and .gz files (only tar.* and zip)

### ISSUE THREAD

**Issue by pbudzon** [ansible/ansible-modules-core#3241]
##### Issue Type:
- Documentation Report
##### Plugin Name:

unarchive
##### Ansible Version:

ansible 2.0.0.2
##### Ansible Configuration:

Default. No changes.
##### Environment:

N/A
##### Summary:

unarchive module only supports tar.\* and .zip files, but the documentation does not mention it and the error message is utterly unhelpful. If you try to use the module with .bz2 or .gz file (no tar), the error message says "Failed to find handler for xxxx. Make sure the required command to extract the file is installed.". The commands are present, it's simply the module which does not support extracting those files.
Either the documentation should be amended (currently it indicates that almost any package can be extracted) or the support for more (commonly used) archivers added.
##### Steps To Reproduce:
1. Compress any file using `bzip2 -c` 
2. Try using the `unarchive` module with this file.
##### Expected Results:

The file would be extracted, similar to running `bzip2 -d`.
##### Actual Results:

```
 fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"backup": null, "content": null, "copy": false, "creates": null, "delimiter": null, "dest": "/tmp/SOURCES/", "directory_mode": null, "follow": false, "force": null, "group": null, "list_files": false, "mode": null, "original_basename": "xxx.gz", "owner": null, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": "/tmp/SOURCES/xxxl.gz"}}, "msg": "Failed to find handler for \"/tmp/SOURCES/xxx.gz\". Make sure the required command to extract the file is installed."}
```

**Comment by sysadmin75** [comment]
I'm unable to reproduce this issue. I tested both gz and bz2 files and both were uncompressed successfully.

Can you provide the task that you use? And maybe provide the source host OS and destination OS?

**Comment by pbudzon** [comment]
Hi,

This is system independent. The problem is obvious when you look at the unarchive module source code - it expects bz2 and gz files to be .tar.bz2 and .tar.gz.
See below:
- `tar -xjf test.tar.bz2` 
- `bzip2 -d test.bz2` 
  Notice the different where the tar was used and where it wasn't. Unarchive module will fail to perform the second decompression.

Tested on Ubuntu 14.04 and Mac OS X.

Complete steps to reproduce on Mac OS X:

```
touch test
bzip2 -c test > test.bz2
mkdir output
ansible localhost -m unarchive -a "src=test.bz2 dest=output"
```

Output:

```
localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Failed to find handler for \"xxxx\". Make sure the required command to extract the file is installed."
}
```

Above tested on Ansible 2.0.2.0.

If you look at the source of the module (https://github.com/ansible/ansible-modules-core/blob/devel/files/unarchive.py), it only works on the following:
- file.zip -`class ZipArchive(object)`
- file.tar.gz or tar.tgz (which are the same thing) - `class TgzArchive(object)`
- file.tar - `class TarArchive(TgzArchive)`
- file.tar.bz2 - `class TarBzipArchive(TgzArchive)`

No support for file.bz2 and file.gz :(

**Comment by kirk-daries** [comment]
Yes please. I ran into the same issue.

+1 for making the error message more meaningful
+1 for bz and gz support

**Comment by EvgenyNikitin** [comment]
vote

**Comment by lasselehtinen** [comment]
Ran to this issue. Vendor supplies the binaries in *-bin.bz2.

**Comment by holybit** [comment]
+1

Ran into same issue w/ `unarchive`:

``` shell
$ ansible --version
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

``` YAML
    - name: Hive client install
      unarchive:
        copy: no
        src: http://mirrors.ocf.berkeley.edu/apache/hive/hive-1.0.1/apache-hive-1.0.1-bin.tar.gz
        dest: /opt
        owner: root
        group: root
```

Above produces following error:

```
TASK [Hive client install] *****************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/tmp/ansible_hGcLW0/apache-hive-1.0.1-bin.tar.gz\". Make sure the required command to extract the file is installed."}
```

**Comment by pbudzon** [comment]
@holybit Are you sure? Unarchive module does extract tar.gz files. The problem is with .gz and .bzip2 files which are not tarred. Do you have tar and gzip installed? :)

**Comment by bronzels** [comment]
@pbudzon this is the output of my system info(centos 7 and tar, gzip, unzip are all installed). and tar xzvf manually is OK to extract. weired even tar.gz can't be extracted. can you pls. help give some hints?

ansible localhost -m unarchive -a "src=/vagrant/hbase-1.1.5-bin.tar.gz dest=/home/hadoop"

localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Failed to find handler for \"/root/.ansible/tmp/ansible-tmp-1464696086.94-264320563741464/source\". Make sure the required command to extract the file is installed."
}

CentOS Linux release 7.0.1406 (Core) 
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
tar (GNU tar) 1.26
Copyright (C) 2011 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
gzip 1.5
Copyright (C) 2007, 2010, 2011 Free Software Foundation, Inc.
Copyright (C) 1993 Jean-loup Gailly.
This is free software.  You may redistribute copies of it under the terms of
the GNU General Public License http://www.gnu.org/licenses/gpl.html.
There is NO WARRANTY, to the extent permitted by law.

Written by Jean-loup Gailly.

**Comment by pbudzon** [comment]
@bronzels This may be a stupid suggestion but... try renaming the file to hbase-1.1.5-bin.**tgz** and see if ansible manages to unpack it?

**Comment by bronzels** [comment]
@pbudzon you are right, i just renamed to hbase-1-1-5, problem solved. looks like a bug in ext name resolving in unarchive module.

**Comment by pbudzon** [comment]
@bronzels Congrats, you found another bug ;)

**Comment by tuxdna** [comment]
I encounter the same issue

```
    - name: Download ElasticSearch
      get_url: url=https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.7.3.tar.gz dest=/home/centos/elasticsearch-1.7.3.tar.gz

    - name: Create ElasticSearch folder
      file: path=/home/centos/elasticsearch-1.7.3 state=directory

    - name: Unzip ElasticSearch
      unarchive: src=/home/centos/elasticsearch-1.7.3.tar.gz dest=/home/centos/elasticsearch-1.7.3 copy=no
```

And I get this error:

```
TASK [Create ElasticSearch folder] *********************************************
changed: [HOST]

TASK [Unzip ElasticSearch] *****************************************************
fatal: [HOST]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/home/centos/elasticsearch-1.7.3.tar.gz\". Make sure the required command to extract the file is installed."}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @setup-elasticsearch.retry

PLAY RECAP *********************************************************************
HOST                 : ok=2    changed=1    unreachable=0    failed=1   
```

Here are additional details:

```
$ ansible --version
ansible 2.1.0.0

$ ansible-playbook --version
ansible-playbook 2.1.0.0
```

HOST details:
- `CentOS 7.1`
- `Linux HOST 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

And all the required binaries for decompression are also present on the HOST ( remote machine )

```
$ which unzip; which tar; which gzip
/usr/bin/unzip
/usr/bin/tar
/usr/bin/gzip
```

**Comment by mowings** [comment]
At the risk of another "me too", same issue here. We have a tar.gz file which worked fine with the 2.0.2.0 unarchive module. We get the missing handler error with 2.1.

**Comment by jdanthine** [comment]
+1

**Comment by DanOfir** [comment]
+1

**Comment by pvitty** [comment]
Same on CentOS 7 with Ansible 2.1.0.0

**Comment by foneox** [comment]
+1 Ubuntu 16.04,  Ansible 2.1

**Comment by CorbanR** [comment]
Also ran into this bug.

**Comment by samohri** [comment]
Upgraded the ansible to  2.1.0.0 and encountered this issue :-( using RHELservers

fatal: [54.66.212.13]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"backup": null, "content": null, "copy": true, "creates": null, "delimiter": null, "dest": "/usr/node/test", "directory_mode": null, "exclude": [], "extra_opts": [], "follow": false, "force": null, "group": null, "keep_newer": false, "list_files": false, "mode": null, "original_basename": "test-javascript-1.76.tar.gz", "owner": null, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": "/root/.ansible/tmp/ansible-tmp-1466411199.18-186704417555180/source"}}, "msg": "Failed to find handler for \"/root/.ansible/tmp/ansible-tmp-1466411199.18-186704417555180/source\". Make sure the required command to extract the file is installed."}
 +1

**Comment by pierreant-p** [comment]
+1

**Comment by adichiru** [comment]
+1 I ran into this bug too on Ubuntu 14.04.4 LTS
ansible 2.1.0.0

achiru@rivendale:~/Projects/origin-ansible$ sha1sum /usr/bin/ansible
754a48999947ac31391d1cc9d59e0934645844cc  /usr/bin/ansible

fatal: [host]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/tmp/apache-maven-3.1.1-bin.tar.gz\". Make sure the required command to extract the file is installed."}

**Comment by benatkin** [comment]
This was a surprising error. I fixed my playbook by switching to `.tar.gz` instead of a plain `.gz`.

**Comment by samohri** [comment]
my file is .tar.gz but still getting this error. Only way i could by pass
this error was using tar command to extract the .tar.gz

Annoying thing is it give you warning saying using unarchive module will be
best option. Will look forward to get this fixed with the team help.

**Comment by jtru** [comment]
After upgrading Ansible from 1.9 to 2.1, the `unarchive` module failed to properly extract an LZO-compressed tarball named `some-archive.tar.lz`, despite the host (Debian Jessie) having both GNU `tar` and `lzop` installed. It worked fine with Ansible 1.9's version of the module.

The exact error produced was:

```
failed: [10.10.10.80] (item={[...]"src": "/root/.ansible/tmp/ansible-tmp-1467126471.85-215507673206574/source", "state": "file", "uid": 0}, "msg": "Failed to find handler for \"/tmp/perl-5.22.2-live.tar.lz\". Make sure the required command to extract the file is installed."}
```

**Comment by jhughes-mc** [comment]
+1

Editing this comment. I ensured gtar, and unzip was installed.

**Comment by jesusfcr** [comment]
Fixed with   
- apt: package=unzip state=present

**Comment by ghost** [comment]
I have the same issue. Using ansible 2.1.0.0, having unzip package doesn't solve the problem.
The file which I tried to extract had 'gz' extension.

**Comment by omame** [comment]
[This thread](https://github.com/ansible/ansible/issues/16078) has a possible explanation.

**Comment by hongxiaolong** [comment]
+1 in CentOS 7.0 ansible 2.1.0.0 with url=http://nginx.org/download/nginx-1.10.1.tar.gz

fatal: [127.0.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to find handler for \"/tmp/ansible_co1DZc/nginx-1.10.1.tar.gz\". Make sure the required command to extract the file is installed."}

_**test tar and error goes with the following msg:**_

`tar -zxvf nginx-1.10.1.tar.gz`

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now

**Comment by radzhome** [comment]
+1, same issue, solution was to use `command: gunzip <file>.gz`

**Comment by dreamcat4** [comment]
+1 On `devel` today.. This sounds like it should be fixed / fixable pretty easily?

Reason:

When unarchive is downloading durectly from remote `http://` url, its only 1 step... no opportunity to rename or anything else. It would be best to fix here, rather than having to have 2 seperate taks being 1st one to `get_url` the file and then the 2nd one to be `command: gunzip` afterwards.

**Comment by ozbillwang** [comment]
+1, run into same issue on centos 7 + ansible 2.1

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by brucellino** [comment]
I hit this problem on ansible 2.2.0.  The suggested fix  - `if not remote_src or copy` -> `if not remote_src and copy` worked.
shipit :ship:

**Comment by Hubbitus** [comment]
It seams work now:

```
- hosts: localhost
  connection: local
  vars:
    tmpdir: /tmp/ansible-unarhive.bug-3241
  tasks:
    - file: name='{{tmpdir}}' state=directory

    - name: Download and unpack tarball
      unarchive:
        src: https://archive.apache.org/dist/apr/apr-0.9.11.tar.bz2
        dest: '{{tmpdir}}'
        copy: no
```

```
PLAY [localhost] ***************************************************************

TASK [file] ********************************************************************
changed: [localhost]

TASK [Download and unpack tarball] *********************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=2    unreachable=0    failed=0   
```

```
$ ansible --version
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

**Comment by dagwieers** [comment]
Ok, this has been brought up before.

First of all, the unarchive module handles archives (as the name suggests) and foo.gz or bar.bz2 are not archives, but compressed files. (That was the default answer in the past #1035) Besides this, the module has more than one problem that requires a pure-python re-implementation. This re-implementation we could call "uncompress" and support compressed files and multiple formats.

Secondly, if the error message is confusing or the documentation needs clarity, please send a pull-request improving the error message. If the PR looks fine, I don't mind asking for it to be merged.

**Comment by caDevOps** [comment]
From the Docs:  http://docs.ansible.com/ansible/unarchive_module.html
"Note
can handle gzip, bzip2 and xz compressed as well as uncompressed tar files"  - A bit misleading, unless i'm just misreading.. Hey that rhymes!

**Comment by dagwieers** [comment]
On Thu, Aug 11, 2016 at 03:46 AM, @caDevOps wrote:

> "Note can handle gzip, bzip2 and xz compressed as well as uncompressed tar files" - A bit misleading, unless i'm just misreading.. Hey that rhymes!

It is factually correct, but could definitely be improved !

BTW Anyone can propose fixes by [editing the file directly](https://github.com/ansible/ansible-modules-core/edit/devel/files/unarchive.py) on GitHub. No prior Git-knowledge required. After editing it automatically clones and asks to create [a pull-request](https://github.com/ansible/ansible-modules-core/pull/4385) for you. So it is super simple to propose fixes that can be applied instantly by the devs.

I admit it are these small improvements (from individual users) that makes Ansible a better experience to new users.

**Comment by dagwieers** [comment]
The **unarchive** module only supports archives, not compressed files. The documentation was improved to make that more clear. We anticipate a new pure-python **uncompress** module that will support compressed files as well as archives, which would then deprecate the unarchive module.

notabug

**Comment by neduma** [comment]
Mak sure it is right and uncorrupted archive file..

```
Archive:  consul_0.8.1_linux_amd64.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of consul_0.8.1_linux_amd64.zip or
        consul_0.8.1_linux_amd64.zip.zip, and cannot find consul_0.8.1_linux_amd64.zip.ZIP, period.
``` 

i had this and i got the same error

**Comment by harobed** [comment]
> We anticipate a new pure-python uncompress module that will support compressed files as well as archives, which would then deprecate the unarchive module.

@dagwieers do you have some news about this feature?

**Comment by dagwieers** [comment]
@harobed Nope, AFAIK nobody is working on it at this time.

**Comment by jmccaughey** [comment]
How is this not a bug? If the archive module supports compression, wouldn't one expect the unarchive module to have decompression?

From the discussion it also sounds like it used to work.

**Comment by J0WI** [comment]
There is a pending PR to implement the uncompress module: #23048

**Comment by ecv** [comment]
(Everything is a file.)[https://en.wikipedia.org/wiki/Everything_is_a_file]

The module being trash and needing reimplementation is a great reason to proceed with `uncompress`, but "archives aren't files" is simply wrong.

**Comment by dagwieers** [comment]
@ecv Nobody said that. Please take your alternative reality somewhere else. What is your agenda ?

**Comment by ecv** [comment]
> First of all, the unarchive module handles archives (as the name suggests) and foo.gz or bar.bz2 are not archives, but compressed files.

My agenda is: I just got hit by this bug, in 2.4, and you guys seem to need help with painting the bikeshed.

**Comment by nethershaw** [comment]
@dagwieers 

> The unarchive module only supports archives, not compressed files. The documentation was improved to make that more clear.

The documentation at http://docs.ansible.com/ansible/latest/unarchive_module.html by no means makes it clear that decompression is unsupported. The string "compress" appears nowhere in the page. It does use the word "unpack" (which in the vernacular _can_ mean "decompress"), and it does reference the extensions `.tar.gz`, `.tar.bz2` and `.tar.xz`, which imply support for gzip, bzip2, and xz (LZMA/LZMA2) compression by their very names.

Furthermore, I _can_ decompress a `.tar.gz` file using the unarchive module in Ansible 2.3.2.0; if it is true that Ansible 2.4.* removes this functionality _before_ the uncompress module becomes available, the loss of functionality should be considered a regression.

**Comment by dagwieers** [comment]
Right, if you have trouble reading words together, here's some help: "compressed files" != "files".

> First of all, the unarchive module handles archives (as the name suggests) and foo.gz or bar.bz2 are not archives, but compressed files. (That was the default answer in the past #1035) ...

Besides, if you would read everything correctly you would not have dropped the reference and you can see that I am simply stating what we were told before I inherited this doomed module. So not only are you not reading incorrectly, you are referencing wrongly and putting words in my mouth. So yes, I am not very happy if someone is doing it.

And that's simply the historical reason for unarchive to not support compressed files. (yes, you see **compressed files**). Does it make sense to add that support ? Not to me, because it is a lost cause and any effort should be reverted elsewhere. Can I stop people from doing it anyway and making a bad module worse, probably not.

Feel free to fix it.

And people that suggest that from the discussions it used to work are also not reading very clearly, as that was about archives, not compressed files. Sorry, I didn't make up that distinction.

**Comment by ecv** [comment]
https://en.wikipedia.org/wiki/Everything_is_a_file

**Comment by dagwieers** [comment]
@nethershaw What loss of functionality ? unarchive **always** only supported compressed archives, never compressed files. If you think the documentation is confusing, feel free to add a PR.

From the documentation notes (you know, if you read to the very end):

> **Note**
> - Requires gtar/unzip command on target host.
> - Can handle .zip files using unzip as well as .tar, .tar.gz, .tar.bz2 and .tar.xz files using gtar.
> - Uses gtar’s --diff arg to calculate if changed or not. If this arg is not supported, it will always unpack the archive.
> - Existing files/directories in the destination which are not in the archive are not touched. This is the same behavior as a normal archive extraction.
> - Existing files/directories in the destination which are not in the archive are ignored for purposes of deciding if the archive should be unpacked or not.
>- For Windows targets, use the win_unzip module instead.

If anyone wants to become the unarchive maintainer, feel free to come forward. If you're not happy with the module, send in a PR. But if you're not happy and think I am responsible, you haven't been investigating very well. I am the bringer of bad news.

**Comment by ecv** [comment]
There we go!

**Comment by dagwieers** [comment]
Ok, if you have an issue with something that worked before in v2.3 and not in v2.4, open a new ticket.
Clearly this is NOT related to this ticket, which is closed.

**Comment by ecv** [comment]
Well I'm still using `command: gunzip` instead of `uncompress` because it doesn't exist yet. So is everyone else in this rather large boat.

You've deprecated existing functionality (as of 2.1!) and not yet shipped the replacement; simply bumping the version from 2.1 to 2.2, 2.3, and 2.4 doesn't fix it or close this issue.

**Comment by dagwieers** [comment]
@ecv This ticket was related to **unarchive** not supporting compressed files. It never did, and unless someone writes it, it never will. The **uncompress** module is future work.

I have not deprecated anything, stop accusing people.

**Comment by nethershaw** [comment]
@dagwieers I believe you are being rather quick to admonish others for not reading when you have not done a complete job of reading me.

I did say "if it is true" that there is a change between Ansible 2.3.2.0 and Ansible 2.4.* with respect to this module -- which is a conditional statement and not to say that I know for certain that there _has_ been a change -- then that would create a feature gap before the proper solution, which I agree _is_ the future uncompress module, is actually available. If you have changed nothing, there is no loss of functionality, and no new gap. That was an invitation for you to confirm the answer to a question that occurred earlier in the thread, not an accusation.

My problem is not with the unarchive module. My problem is with the documentation, which you claimed to have made more clear, and which _I did read to the end_ (and from which I copied the list of file extensions verbatim). The documentation makes no negative statements to the effect of what does _not constitute an archive_, which is a subtle one, and one that you have admirably fought to make clear in this comment stream, but which you have not in fact placed in the documentation. A mere mention there of the fact that a gzipped file that hasn't been passed through `tar` first, for example, is not technically an archive, would go far. Hence the popular confusion.

If you would kindly add to the documentation what you have gone to considerable effort to say here, it would dispose of the need for this banter.

**Comment by dagwieers** [comment]
@nethershaw The module documentation gives a list of extensions that are supported. That's what the change was about. https://github.com/ansible/ansible-modules-core/pull/4385

If you know how to improve the documentation, great ! Send in a PR with the wordings that make the documentation better. I will happily merge your improvements.

**Comment by dreamcat4** [comment]
I finally understand now what this disagreement seems to be about. OK then someone is being a bit OCD about this....

Because while you *could* say that an archive isn't a compressed file, actually the mere act of compressing something is also just another way of archiving something. And just because compressed files are compressed files doesn't actually mean they are not considered something else too: i.e. archives.

It really just depends about how strict or lax you are being with the term 'archive'. You are saying that the word archive in computing has a very specific mapping to certain file types. And that a .gz can never be considered as an archive of only 1 file. Well I wasn't ever aware that there was an n-ary requirement for archives that make them only applicable to multiple files... and can you see how this is all getting pretty silly now?

Anyway please look this / why the very practical reason for why I want this feature myself, and I'm sure other people here will agree with the logic of it:

https://github.com/ansible/ansible-modules-core/issues/3241#issuecomment-234589729

**Comment by dreamcat4** [comment]
BTW here is an official wikipedia article / page showing specifically why the author here is flat-out wrong in not considering compressed files as archives:

https://en.wikipedia.org/wiki/Archive_file#Types

Sorry, but you're just plain wrong mate.

**Comment by dagwieers** [comment]
@dreamcat4 I don't think it is about OCD or strict interpretation. The module was created to work with zip-files and (compressed) tar-files because it only drives **unzip** and **gtar**. That's why it was named unarchive. (do a `man zip` or `man gzip` and you'll see that unzip will talk about ZIP archives and gzip will talk about compressed files).

And it's not going to be easy to add compressed-file support because you will have to rewrite the whole module (and likely break some stuff in the meantime, getting cursed at, etc.). And nobody is willing to do that (and in my opinion it's a lost effort). So people complain loudly because they need something, nobody is interested of doing the effort.

The solution to create an **uncompress** module that handles only compressed files is making things worse, because people don't understand the difference. So the only solution here is the pure-python uncompress module that supports both compressed files and (compressed) archives from the very start.

More information here: https://github.com/ansible/community/wiki/Module:-unarchive

PS I contemplated to create a **7zip** and **win_7zip** module that's just a wrapper around 7zip. It's going to be a rude non-idempotent implementation, but the upside is that it would simply work.

**Comment by dagwieers** [comment]
Ok, I am fed up with this. If people want to improve the documentation, go ahead. I gave zero interests in a semantic discussion with a bunch of people that are not contributing to a solution and only complain.

**Comment by dagwieers** [comment]
I officially stop being the maintainer of the **unarchive** module.
Congratulations, now the module is unmaintained.

Good luck to whoever wants to take become the new maintainer.
